File: booksy-api/src/auth/auth.controller.ts
import { Controller, Post, Get, Body, Param, HttpStatus, HttpException } from '@nestjs/common';
import { AuthService } from './auth.service';

@Controller('api/auth')
export class AuthController {
  constructor(private readonly authService: AuthService) {}

  // registro
  @Post('signup')
  async register(@Body() body: { name: string; email: string; password: string }) {
    const { name, email, password } = body;

    // validar si email ya existe
    const exists = await this.authService.emailExists(email);
    if (exists) {
      throw new HttpException({ error: 'email ya existe' }, HttpStatus.FORBIDDEN);
    }

    // crear usuario
    const user = await this.authService.createUser(name, email, password);

    return {
      authToken: `token_${user._id}`,
      userId: user._id,
    };
  }

  // login
  @Post('login')
  async login(@Body() body: { email: string; password: string }) {
    const { email, password } = body;

    // buscar usuario
    const user = await this.authService.findByEmail(email);

    if (!user || user.password !== password) {
      throw new HttpException({ error: 'credenciales invalidas' }, HttpStatus.FORBIDDEN);
    }

    return {
      authToken: `token_${user._id}`,
      userId: user._id,
    };
  }

  // obtener usuario
  @Get('me/:userId')
  async getUser(@Param('userId') userId: string) {
    const user = await this.authService.findById(userId);

    if (!user) {
      throw new HttpException({ error: 'usuario no encontrado' }, HttpStatus.NOT_FOUND);
    }

    return {
      id: user._id,
      name: user.name,
      email: user.email,
    };
  }
}

File: booksy-api/src/main/java/com/booksy/backend/controller/AuthController.java
package com.booksy.backend.controller;

import com.booksy.backend.model.User;
import com.booksy.backend.repository.UserRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.HashMap;
import java.util.Map;
import java.util.Optional;

@RestController
@RequestMapping("/api/auth")
@CrossOrigin(origins = "*")
public class AuthController {

    @Autowired
    private UserRepository userRepository;

    // registro
    @PostMapping("/signup")
    public ResponseEntity<?> register(@RequestBody Map<String, String> request) {
        String name = request.get("name");
        String email = request.get("email");
        String password = request.get("password");

        // validar si el email ya existe
        if (userRepository.existsByEmail(email)) {
            return ResponseEntity.status(HttpStatus.FORBIDDEN).body(Map.of("error", "email ya existe"));
        }

        // crear usuario
        User user = new User(name, email, password);
        User savedUser = userRepository.save(user);

        // respuesta
        Map<String, Object> response = new HashMap<>();
        response.put("authToken", "token_" + savedUser.getId());
        response.put("userId", savedUser.getId());

        return ResponseEntity.ok(response);
    }

    // login
    @PostMapping("/login")
    public ResponseEntity<?> login(@RequestBody Map<String, String> request) {
        String email = request.get("email");
        String password = request.get("password");

        // buscar usuario
        Optional<User> userOpt = userRepository.findByEmail(email);

        if (userOpt.isEmpty() || !userOpt.get().getPassword().equals(password)) {
            return ResponseEntity.status(HttpStatus.FORBIDDEN).body(Map.of("error", "credenciales invalidas"));
        }

        User user = userOpt.get();

        // respuesta
        Map<String, Object> response = new HashMap<>();
        response.put("authToken", "token_" + user.getId());
        response.put("userId", user.getId());

        return ResponseEntity.ok(response);
    }

    // obtener usuario
    @GetMapping("/me/{userId}")
    public ResponseEntity<?> getUser(@PathVariable String userId) {
        Optional<User> userOpt = userRepository.findById(userId);

        if (userOpt.isEmpty()) {
            return ResponseEntity.status(HttpStatus.NOT_FOUND).body(Map.of("error", "usuario no encontrado"));
        }

        User user = userOpt.get();

        Map<String, Object> response = new HashMap<>();
        response.put("id", user.getId());
        response.put("name", user.getName());
        response.put("email", user.getEmail());

        return ResponseEntity.ok(response);
    }
}